lms/djangoapps/mfe_config_api/views.py

```diff
@@ -0,0 +1,51 @@
+"""
+MFE API Views for useful information related to mfes.
+"""
+
+from django.conf import settings
+from django.http import HttpResponseNotFound, JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_page
+from rest_framework import status
+from rest_framework.views import APIView
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+
+class MFEConfigView(APIView):
+    """
+    Provides an API endpoint to get the MFE_CONFIG from site configuration.
+    """
+
+    @method_decorator(cache_page(settings.MFE_CONFIG_API_CACHE_TIMEOUT))
+    def get(self, request):
+        """
+        GET /api/v1/mfe_config
+        or
+        GET /api/v1/mfe_config?mfe=name_of_mfe
+
+        **GET Response Values**
+        ```
+        {
+            "BASE_URL": "https://name_of_mfe.example.com",
+            "LANGUAGE_PREFERENCE_COOKIE_NAME": "example-language-preference",
+            "CREDENTIALS_BASE_URL": "https://credentials.example.com",
+            "DISCOVERY_API_BASE_URL": "https://discovery.example.com",
+            "LMS_BASE_URL": "https://courses.example.com",
+            "LOGIN_URL": "https://courses.example.com/login",
+            "LOGOUT_URL": "https://courses.example.com/logout",
+            "STUDIO_BASE_URL": "https://studio.example.com",
+            "LOGO_URL": "https://courses.example.com/logo.png"
+        }
+        ```
+        """
+
+        if not settings.ENABLE_MFE_CONFIG_API:
+            return HttpResponseNotFound()
+
+        mfe_config = configuration_helpers.get_value('MFE_CONFIG', {})
+        if request.query_params.get('mfe'):
+            mfe = str(request.query_params.get('mfe')).upper()
+            mfe_config.update(configuration_helpers.get_value(f'MFE_CONFIG_{mfe}', {}))
```

Comment:
_If you have already addressed this, please let me know._
I had a similar situation in another PR that introduced changes related to site configurations, increasing the technical debt given that a [deprecation ticket](https://github.com/openedx/platform-roadmap/issues/21) already exists. My approach there was to suggest replacing site configurations with another type of configuration strategy. I commented on [this approach](https://openedx.slack.com/archives/CGB0S3L12/p1655937179658019) in the Deprecation slack channel.
Do you think that suggestion is suitable in this context?

Reply:
I got it. It's great that it's in the ADR. But still, if later you're going to use Django settings, what is the advantage of using site configurations now? Is the behavior going to change then? Can you default getting site configurations to Django settings, so using site configs is not required? Please, let me know if this even makes sense.

Reply:
Using site configuration was the main idea, [see the proposal document](https://docs.google.com/document/d/1-FHIQmyeQZu3311x8eYUNMru4JX7Yb3UlqjmJxvM8do/edit?usp=sharing); but I already made changes in the PR to also support Django settings.